pkg/console/util.go

```diff
@@ -1041,109 +1038,24 @@ func generateEnvAndConfig(g *gocui.Gui, hvstConfig *config.HarvesterConfig) ([]s
 	return env, elementalConfig, nil
 }
 
-// internal objects to parse lsblk output
-type BlockDevices struct {
-	Disks []Device `json:"blockdevices"`
-}
-
-type Device struct {
-	Name     string   `json:"name"`
-	Size     string   `json:"size"`
-	DiskType string   `json:"type"`
-	WWN      string   `json:"wwn,omitempty"`
-	Serial   string   `json:"serial,omitempty"`
-	Label    string   `json:"label,omitempty"`
-	Children []Device `json:"children,omitempty"`
-}
-
-func generateDiskEntry(d Device) string {
-	return fmt.Sprintf("%s %s", d.Name, d.Size)
-}
-
-const (
-	diskType = "disk"
-)
-
-// identifyUniqueDisks parses the json output of lsblk and identifies
-// unique disks by comparing their serial number info and wwn details
-func identifyUniqueDisks(output []byte) ([]string, error) {
-
-	resultMap, err := filterUniqueDisks(output)
-	if err != nil {
-		return nil, err
-	}
-
-	returnDisks := make([]string, 0, len(resultMap))
-	// generate list of disks
-	for _, v := range resultMap {
-		returnDisks = append(returnDisks, generateDiskEntry(v))
-	}
-
-	return returnDisks, nil
-}
-
-// filterUniqueDisks will dedup results of disk output to generate a map[disName]Device of unique devices
-func filterUniqueDisks(output []byte) (map[string]Device, error) {
-	disks := &BlockDevices{}
-	err := json.Unmarshal(output, disks)
-	if err != nil {
-		return nil, fmt.Errorf("error unmarshalling lsblk json output: %v", err)
-	}
-	// identify devices which may be unique
-	dedupMap := make(map[string]Device)
-	for _, disk := range disks.Disks {
-		if disk.DiskType == diskType {
-			// no serial or wwn info present
-			// add to list of disks
-			if disk.WWN == "" && disk.Serial == "" {
-				dedupMap[disk.Name] = disk
-				continue
-			}
-
-			if disk.Serial != "" {
-				_, ok := dedupMap[disk.Serial]
-				if !ok {
-					dedupMap[disk.Serial] = disk
-				}
-			}
-
-			// disks may have same serial number but different wwn when used with a raid array
-			// as evident in test data from a host with a raid array
-			// in this case if serial number is same, we still check for unique wwn
-			if disk.WWN != "" {
-				_, ok := dedupMap[disk.WWN]
-				if !ok {
-					dedupMap[disk.WWN] = disk
-				}
-				continue
-			}
-		}
-	}
-	// devices may appear twice in the map when both serial number and wwn info is present
-	// we need to ensure only unique names are shown in the console
-	resultMap := make(map[string]Device)
-	for _, v := range dedupMap {
-		resultMap[v.Name] = v
-	}
-	return resultMap, nil
-}
-
 // getDiskOptions identifies disks that can be used for installation
 func getDiskOptions() ([]widgets.Option, error) {
-	output, err := exec.Command("/bin/sh", "-c", `lsblk -J -o NAME,SIZE,TYPE,WWN,SERIAL,LABEL`).CombinedOutput()
+	disks, err := util.GetUniqueDisks()
 	if err != nil {
 		return nil, err
 	}
 
-	disks, err := identifyUniqueDisks(output)
+	if len(disks) == 0 {
+		return nil, fmt.Errorf("got 0 disk")
```

Comment:
return error like 
`Harvester needs at least one disk to install the system, but no disk is found` ?